csharp/autobuilder/Semmle.Autobuild.Shared/BuildTools.cs

```diff
@@ -51,12 +49,13 @@ public static IEnumerable<VcVarsBatFile> GetCandidateVcVarsFiles(IBuildActions a
                         {
                             if (majorVersion < 15)
                             {
-                                yield return new VcVarsBatFile(actions.PathCombine(vsInstallation.InstallationPath, @"VC\vcvarsall.bat"), majorVersion, "x86");
+                                yield return new VcVarsBatFile(actions.PathCombine(vsInstallation.InstallationPath, @"VC\vcvarsall.bat"), majorVersion);
                             }
                             else
                             {
-                                yield return new VcVarsBatFile(actions.PathCombine(vsInstallation.InstallationPath, @"VC\Auxiliary\Build\vcvars32.bat"), majorVersion, "x86");
-                                yield return new VcVarsBatFile(actions.PathCombine(vsInstallation.InstallationPath, @"VC\Auxiliary\Build\vcvars64.bat"), majorVersion, "x64");
+                                yield return new VcVarsBatFile(actions.PathCombine(vsInstallation.InstallationPath, @"Common7\Tools\VsDevCmd.bat"), majorVersion);
+                                yield return new VcVarsBatFile(actions.PathCombine(vsInstallation.InstallationPath, @"VC\Auxiliary\Build\vcvars32.bat"), majorVersion);
```

Comment:
If `vcvars*` exist, shouldn't we try them first, since they should be a superset of `VsDevCmd.bat`? For example, if you had some C++ projects in your solution, you might want the C++ paths set up so your build succeeds, even if you're only extracting C#.

Reply:
Makes sense, given that this code is also used by the C++ autobuilder on Windows.